accessor/accessor_helper.hpp

```diff
@@ -78,13 +78,16 @@ struct row_major_helper_s {
         const std::array<SizeType, (total_dim > 1 ? total_dim - 1 : 0)>& stride,
         IndexType first, Indices&&... idxs)
     {
+        auto size_access = reinterpret_cast<const SizeType*>(&size);
```

Comment:
These `reinterpret_cast`s are not great...
Why don't you simply use `size.data()`? This is the safer approach as it is (to my knowledge) not guaranteed that `reinterpret_cast` works fine with `array`s.
Also, why is this necessary? `std::array` has been around for quite a while, so it should work. Additionally, making the accesses less safe / less readable and potentially remove optimizations just for one debug builds seems unreasonable.